web-wallet/src/lib/containers/MigrateContract/MigrateContract.svelte

```diff
@@ -97,23 +154,9 @@
     return () => resizeObserver.disconnect();
   });
 
-  /**
-   * @param {Number} ms
-   */
-  function delay(ms) {
-    return new Promise((resolve) => setTimeout(resolve, ms));
-  }
-
-  function resetMigrationIfConnected() {
-    if (isWalletConnected) {
-      isWalletConnected = false;
-      connectedWalletAddress = "";
-      connectedWalletBalance = 10;
-      amount = undefined;
-      isMigrationInitialized = false;
-      isMigrationBeingApproved = false;
-    }
-  }
+  onDestroy(async () => {
+    await walletDisconnect();
```

Comment:
do we really need to await this?

Reply:
According to the docs, we need to await the disconnect.